master/usage/policy/secure-metrics.md

```diff
@@ -0,0 +1,408 @@
+---
+title: Securing access to Calico's metrics endpoints
+---
+
+## About securing access to {{site.prodname}}'s metrics endpoints
+
+When using {{ site.prodname }} with prometheus metrics enabled, it may be desirable to use network policy
+to limit access to {{ site.prodname }}'s metrics endpoints.
+
+### Purpose of this page
+
+Provide example workflows for creating network policies to limit access to {{site.prodname}}'s metrics endpoints.
+
+## Prerequisites
+
+- {{site.prodname}} is installed with [prometheus metrics reporting enabled](TODO).
+- calicoctl is [installed in your PATH and configured to access the data store](../calicoctl/install).
+
+## Using a blacklist approach
+
+This approach allows all traffic to your hosts by default, but lets you limit access to specific ports using
+Calico policy.
+
+### Overview
+
+The basic process is as follows:
+
+1. Create a default network policy which allows traffic to and from your hosts.
+1. Create host endpoints for each node which you'd like to secure.
+1. Create a network policy which denies unwanted traffic to the {{site.prodname}} metrics endpoints.
+1. Apply labels to allow access to the prometheus metrics.
+
+### Example for calico/node
+
+This example shows how to limit access to the {{site.nodecontainer}} prometheus metrics endpoints.
+
+1. Create a default network policy to allow host traffic
+
+   By creating host endpoints in the next steps, we will restrict traffic to and from the interfaces listed
+   on each of {{site.prodname}} hosts. To ensure we don't break communication with the hosts, we'll create a default network policy
+   which will allow all traffic to and from the host endpoints.
+
+   To do this, create a file named `default-host-policy.yaml` with the following contents.
+
+   ```yaml
+   apiVersion: projectcalico.org/v3
+   kind: GlobalNetworkPolicy
+   metadata:
+     name: default-host
+   spec:
+     # Select all {{site.prodname}} nodes.
+     selector: running-calico == "true"
+     order: 5000
+     ingress:
+     - action: Allow
+     egress:
+     - action: Allow
+   ```
+
+   Then, use calicoctl to apply this policy.
+
+   ```shell
+   calicoctl apply -f default-host-policy.yaml
+   ```
+
+1. List the nodes on which {{site.prodname}} is running. In this case, we have two nodes in the cluster.
+
+   ```shell
+   $ calicoctl get nodes
+   NAME
+   kubeadm-master
+   kubeadm-node-0
+   ```
+
+1. Create host endpoints for each {{site.prodname}} node.
+
+   Create a file named `host-endpoints.yaml` containing a host endpoint for each node listed
+   above. In this example, the contents would look like this.
+
+   ```yaml
+   apiVersion: projectcalico.org/v3
+   kind: HostEndpoint
+   metadata:
+     name: kubeadm-master.eth0
+     labels:
+       running-calico: "true"
+   spec:
+     interfaceName: eth0
+     node: kubeadm-master
```

Comment:
Should we encourage users to supply the expected IPs too?  I'm worried that we don't want to encourage bad practice (and policy works much better if you have those IPs in place).

Reply:
Maybe, although it makes it substantially harder from an operational standpoint.